### ISSUE TITLE
OSLC Link Discovery Management - shall owl:sameAs be considered

### ISSUE THREAD

**Issue by MartinUlrich** [oslc-op/oslc-specs#599]
Via the use of owl:sameAs synonyms e.g. for predicates can be defined.

LDM could make use of that feature and not only return the links for the requested predicates but also the links for the synonyms.

Proposal is **not to support synonyms defined via owl:sameAS**.

Rationale:
- LDM should not know anything about the semantics of the predicates, to keep the service simple
- semantics of predicates can easily be managed by the clients

**Comment by berezovskyi** [comment]
I agree, while not sharing the rationales you listed. Instead, `owl:sameAs` has quite particular semantantics (mathematically) and is widely known to be misused. Cf. http://events.linkeddata.org/ldow2010/papers/ldow2010_paper09.pdf and https://stackoverflow.com/questions/7073229/can-we-use-owlsameas-between-two-predicates regarding the semantics.

Additionally, for the inference of equivalent properties to work, a reasoner must be provided with the appropriate ontological statements. In our case, it means that the OSLC OP shall amend the standards in question to add RDFS and/or OWL statements with the indended semantics.

If you are interested in looking closer at this, I would appreciate the use case for "synonyms" to be detailed a bit more. This way, I will be able to suggest the appropriate mechanism (in my opinion). Generally speaking, we should try to evaluate the reasoning modes in the increasing order of implementation complexity:

1. RDFS Simple (as defined by [Jena](https://jena.apache.org/documentation/inference/#RDFSconfiguration)),
2. RDFS Default,
3. OWL Mini (similarly, defined by [Jena](https://jena.apache.org/documentation/inference/#OWLcoverage); sadly, it's not aligned with the OWL Horst subset or the OWL2 EL, QL, RL, DL profiles; I chose Mini as a starting point instead of Micro as I haven't seen many ontologies that don't rely on at least some `owl:disjointWith` constraints)
4. ...

Note that OWL QL, for example, only allows individual equality assertions (`oslc_rm:proj1_RequirementCollection123 owl:sameAs oslc_rm:proj1_import_RequirementCollection123`) but not the other forms.

**Comment by MartinUlrich** [comment]
I agree and I pull back the rationale: semantics of predicates can easily be managed by the clients

Going into the direction of reasoning to me is a new thread - or even a new universe.
Just one note to it, I can hardly imaging how configuration aware links would benefit from applying reasoning means.

**Comment by DavidJohnHoney** [comment]
owl:sameAs support should be regarded as mandatory. A very common pattern in IBM Rational ELM is to use system generated predicates for the link themselves as these are specific to project areas. Each system-generated predicate that has shared semantics will contribute a statement of the form:
    `    <propertyDefinitionUri> owl:sameAs <stableExternalUri>`
See https://jazz.net/wiki/bin/view/LinkedData/BestPractices for details.
